Fixes from upstream 3.2 for Arista devices

samdoran opened the pull request.

dotwaffle commented:
Not to doubt you, Sam (you're doing a fantastic job!) but this is something lots of people use. Can you confirm your version works with a live piece of kit? Once you confirm, I'll merge.

As an aside, I'm looking at giving up maintainership of this fork... Is it something you'd be interested in taking over?

samdoran commented:
Not at all a problem! I only have two Arista devices to test with, so I'm not sure if it's an adequate test. These devices are new to me, which is why I discovered this bug.

This is the error I was seeing (I added the `<removed>`):
```
arrancid -d <removed>
executing clogin -t 90 -c"show version;show boot-config;show env all;dir flash:;show inventory;show boot-extensions;show extensions;show running-config" <removed>
PROMPT MATCH: <removed>.14:35:20#
HIT COMMAND:<removed>.14:35:20#show version
    In ShowVersion: <removed>.14:35:20#show version
HIT COMMAND:<removed>.14:35:20#show boot-config
    In ShowBoot: <removed>.14:35:20#show boot-config
HIT COMMAND:<removed>.14:35:20#show env all
    In ShowEnv: <removed>.14:35:20#show env all
<removed>: missed cmd(s): dir flash:,show inventory,show extensions,show boot-extensions,show running-config
<removed>: missed cmd(s): dir flash:,show inventory,show extensions,show boot-extensions,show running-config
<removed>: End of run not found
<removed>: End of run not found
```

After making the changes in 3ecd8ad (which is from RANCID 3.2), `arrancid` now runs without error.

Debug output showing all the commands successfully run:
```
arrancid -dl <removed hostname>
executing clogin -t 90 -c"show version;show boot-config;show env all;dir flash:;show inventory;show boot-extensions;show extensions;diff startup-config running-config;show running-config" <removed hostname>
executing clogin -t 90 -c"show version;show boot-config;show env all;dir flash:;show inventory;show boot-extensions;show extensions;diff startup-config running-config;show running-config" <removed hostname>
PROMPT MATCH: <removed hostname>.\d+:\d+:\d+#
HIT COMMAND:<removed hostname>.14:16:40#show version
    In ShowVersion: <removed hostname>.14:16:40#show version
HIT COMMAND:<removed hostname>.14:16:40#show boot-config
    In ShowBoot: <removed hostname>.14:16:40#show boot-config
HIT COMMAND:<removed hostname>.14:16:40#show env all
    In ShowEnv: <removed hostname>.14:16:40#show env all
HIT COMMAND:<removed hostname>.14:16:41#dir flash:
    In ShowFlash: <removed hostname>.14:16:41#dir flash:
HIT COMMAND:<removed hostname>.14:16:41#show inventory
    In ShowInventory: <removed hostname>.14:16:41#show inventory
HIT COMMAND:<removed hostname>.14:16:41#show boot-extensions
    In ShowBootExt: <removed hostname>.14:16:41#show boot-extensions
HIT COMMAND:<removed hostname>.14:16:41#show extensions
    In ShowExt: <removed hostname>.14:16:41#show extensions
HIT COMMAND:<removed hostname>.14:16:41#diff startup-config running-config
    In DiffConfig: <removed hostname>.14:16:41#diff startup-config running-config
HIT COMMAND:<removed hostname>.14:16:42#show running-config
    In WriteTerm: <removed hostname>.14:16:42#show running-config
```

Here is a snippet of the output (I added the `<removed>`):
```
!RANCID-CONTENT-TYPE: arista
!
!Model: Arista DCS-7150S-24-R
!Hardware version: <removed>
!Serial number: <removed>
!System MAC address: <removed>
!Software image version: <removed>
!Architecture: i386
!Internal build version: <removed>
!Internal build ID: <removed>
!Total memory: 4019184 kB
!
!Software image: flash:/EOS-<removed>
!Console speed: (not set)
!Aboot password (encrypted): (not set)
!Memory test iterations: (not set)
```

dotwaffle commented:
I'm so sorry, I completely missed this! Merged!
